src/main/java/com/kine/Kine/food/controller/FoodController.java

```diff
@@ -0,0 +1,102 @@
+package com.kine.Kine.food.controller;
+import com.kine.Kine.food.dto.CreateFoodDTO;
+import com.kine.Kine.food.dto.GetFoodDTO;
+import com.kine.Kine.food.dto.UpdateDTO;
+import com.kine.Kine.food.exception.*;
+import com.kine.Kine.food.service.FoodService;
+import com.kine.Kine.food.dto.FoodDTO;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.data.domain.Pageable;
+
+
+
+@RestController
+@RequestMapping("/api/food")
+public class FoodController {
+    private final FoodService foodService;
+    public FoodController(FoodService foodService){
+        this.foodService = foodService;
+    }
+
+
+    @GetMapping(value = "/{id}")
+    public ResponseEntity<?> searchFoodById(@PathVariable(value = "id") Long id) {
+        try {
+            GetFoodDTO foodDTO = foodService.findFoodById(id);
+            return ResponseEntity.ok(foodDTO); // Return OK status with the found resource
+        } catch (RessourceNotFoundException e) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                    .body(e.getMessage()); // Return meaningful response body
+        } catch (Exception e) {
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
+                    .body(e.getMessage()); // Handle other exceptions
+        }
+    }
+
+
+    @GetMapping
+    public ResponseEntity<?> searchFood(@RequestParam(value = "name", required = false) String name,
+                                                       @RequestParam(defaultValue = "0") int page,
+                                                       @RequestParam(defaultValue = "10") int size) {
+        try {
+            Pageable pageable = PageRequest.of(page, size);
+            if (name != null) {
+                return ResponseEntity.status(HttpStatus.OK).body(foodService.findFoodByName(name, pageable));
+            } else {
+                return ResponseEntity.status(HttpStatus.FOUND).body(foodService.getFood(pageable));
+            }
+        }catch (Exception e){
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
+                    .body(e.getMessage());
+        }
+
+    }
+
+
+    @PostMapping
+    public ResponseEntity<?> createFood(@RequestBody CreateFoodDTO createFoodDTO) {
+        try{
+            return ResponseEntity.status(HttpStatus.CREATED).body(foodService.createFood(createFoodDTO));
+        }catch (RessourceAlreadyExistException e){
+            return new ResponseEntity<>(e.getMessage(), HttpStatus.CONFLICT);
```

Comment:
1. Formatting
2. Internal Error
3. Use the previous ResponseEntity.status().body() not this one